v2/reader.go

```diff
@@ -9,13 +9,30 @@ import (
 
 	"github.com/ipfs/go-cid"
 	"github.com/ipld/go-car/v2/internal/carv1"
+	"golang.org/x/exp/mmap"
 )
 
 // Reader represents a reader of CAR v2.
 type Reader struct {
 	Header Header
 	r      io.ReaderAt
 	roots  []cid.Cid
+	closer io.Closer
+}
+
+func NewReaderFromCARv2File(path string) (*Reader, error) {
+	f, err := mmap.Open(path)
+	if err != nil {
+		return nil, err
+	}
+
+	r, err := NewReader(f)
+	if err != nil {
+		return nil, err
+	}
+
+	r.closer = f
```

Comment:
I think we should remove this and use a type assertion on `r.r` being an `io.Closer` instead. That way there's less manual magic, and Close automatically does the right thing regardless of the API that was used to create the Reader.

Reply:
To clarify, I mean the type assertion should be inside the Close method.